### PR TITLE
Added PackedRasterCanvas

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Moved code specific to packed raster images to separate `PackedRasterCanvas`.
@marcusnaslund

### CODE REVIEW

File: source/draw/RasterCanvas.ooc [hunk 1]
```diff
@@ -71,59 +71,4 @@ RasterCanvas: abstract class extends Canvas {
...
-	_blendSquare: static func (sourceBuffer, resultBuffer: UInt8*, sourceStride, resultStride, sourceRow, sourceColumn, row, column: Int, weightTopLeft, weightTopRight, weightBottomLeft, weightBottomRight: Float, bytesPerPixel: Int) {
-		finalValue: UInt8 = 0
-		for (i in 0 .. bytesPerPixel) {
-			finalValue = weightTopLeft > 0.0f ? weightTopLeft * sourceBuffer[sourceColumn * bytesPerPixel + sourceRow * sourceStride + i] : 0
-			finalValue += weightTopRight > 0.0f ? weightTopRight * sourceBuffer[(sourceColumn + 1) * bytesPerPixel + sourceRow * sourceStride + i] : 0
-			finalValue += weightBottomLeft > 0.0f ? weightBottomLeft * sourceBuffer[sourceColumn * bytesPerPixel + (sourceRow + 1) * sourceStride + i] : 0
-			finalValue += weightBottomRight > 0.0f ? weightBottomRight * sourceBuffer[(sourceColumn + 1) * bytesPerPixel + (sourceRow + 1) * sourceStride + i] : 0
-			resultBuffer[column * bytesPerPixel + row * resultStride + i] = finalValue
-		}
-	}
```
Comment (sebastianbaginski) [inline source/draw/RasterCanvas.ooc]: This code is moved to `PackedRasterCanvas` in unchanged form.

File: source/draw/RasterPacked.ooc [hunk 2]
```diff
@@ -25,6 +25,67 @@ import RasterBgra
 import RasterBgr
 import RasterMonochrome
 import Image
+import Canvas, RasterCanvas
+
+PackedRasterCanvas: abstract class extends RasterCanvas {
+	target ::= this _target as RasterPacked
```
Comment (sebastianbaginski) [inline source/draw/RasterPacked.ooc]: I think this will help to simplify the code in next PRs.

File: source/draw/RasterPacked.ooc [hunk 3]
```diff
@@ -25,6 +25,67 @@ import RasterBgra
...
+		finalValue: UInt8 = 0
+		for (i in 0 .. bytesPerPixel) {
+			finalValue = weightTopLeft > 0.0f ? weightTopLeft * sourceBuffer[sourceColumn * bytesPerPixel + sourceRow * sourceStride + i] : 0
+			finalValue += weightTopRight > 0.0f ? weightTopRight * sourceBuffer[(sourceColumn + 1) * bytesPerPixel + sourceRow * sourceStride + i] : 0
+			finalValue += weightBottomLeft > 0.0f ? weightBottomLeft * sourceBuffer[sourceColumn * bytesPerPixel + (sourceRow + 1) * sourceStride + i] : 0
+			finalValue += weightBottomRight > 0.0f ? weightBottomRight * sourceBuffer[(sourceColumn + 1) * bytesPerPixel + (sourceRow + 1) * sourceStride + i] : 0
+			resultBuffer[column * bytesPerPixel + row * resultStride + i] = finalValue
+		}
+	}
+}
```
Comment (sebastianbaginski) [inline source/draw/RasterPacked.ooc]: Code moved from `RasterCanvas`, unchanged.

**Comment by marcusnaslund** [comment]
Please fine someone else to peer review this one.

**Comment by sebastianbaginski** [comment]
OK. @emilwestergren can you take a look ? It only moves some code around.

File: source/draw/RasterMonochrome.ooc [hunk 4]
```diff
@@ -25,7 +25,7 @@ import Image, FloatImage
 import Color
 import Canvas, RasterCanvas
 
-MonochromeRasterCanvas: class extends RasterCanvas {
+MonochromeRasterCanvas: class extends PackedRasterCanvas {
```
Comment (emilwestergren) [inline source/draw/RasterMonochrome.ooc]: We should set a naming standard here (`MonochromeRasterCanvas` was my fault I think).
`RasterImage` -> `RasterPacked` -> `RasterMonochrome`
`RasterCanvas` -> `RasterCanvasPacked` -> `RasterCanvasMonochrome`

What do you propose?

Comment (sebastianbaginski) [inline source/draw/RasterMonochrome.ooc]: Yeah I agree, I was just following the existing convention `<Type>RasterCanvas` :) I think both `Raster<Type>Canvas` and `RasterCanvas<Type>` are ok.

Comment (emilwestergren) [inline source/draw/RasterMonochrome.ooc]: Yes maybe `RasterMonochromeCanvas` is better. Can you fix this in a new PR after this?

Comment (sebastianbaginski) [inline source/draw/RasterMonochrome.ooc]: Sure.

Comment (emilwestergren) [inline source/draw/RasterMonochrome.ooc]: https://github.com/cogneco/ooc-kean/issues/974

Comment (sebastianbaginski) [inline source/draw/RasterMonochrome.ooc]: Thanks

**Pull request merged**